app/uk/gov/hmrc/alcoholdutyreturns/config/AppConfig.scala

```diff
@@ -17,28 +17,46 @@
 package uk.gov.hmrc.alcoholdutyreturns.config
 
 import play.api.Configuration
+import uk.gov.hmrc.alcoholdutyreturns.models.ReturnId
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
+import java.text.MessageFormat
 import javax.inject.{Inject, Singleton}
 
 @Singleton
 class AppConfig @Inject() (config: Configuration, servicesConfig: ServicesConfig) {
-
   val appName: String = config.get[String]("appName")
 
   private lazy val adrAccountHost: String =
     servicesConfig.baseUrl("alcohol-duty-accounts")
 
+  private lazy val returnsHost: String =
+    servicesConfig.baseUrl("returns")
+
+  lazy val returnsClientId           = getConfStringAndThrowIfNotFound("returns.clientId")
+  lazy val returnsSecret             = getConfStringAndThrowIfNotFound("returns.secret")
+  lazy val returnsGetReturnUrlFormat = new MessageFormat(getConfStringAndThrowIfNotFound("returns.url.getReturn"))
+
   val dbTimeToLiveInSeconds: Int = config.get[Int]("mongodb.timeToLiveInSeconds")
 
   def getSubscriptionSummaryUrl(appaId: String): String =
     s"$adrAccountHost/alcohol-duty-account/subscriptionSummary/$appaId"
 
-  def getOpenObligationDataUrl(appaId: String, periodKey: String) =
-    s"$adrAccountHost/alcohol-duty-account/openObligationDetails/$appaId/$periodKey"
+  def getOpenObligationDataUrl(returnId: ReturnId) =
+    s"$adrAccountHost/alcohol-duty-account/openObligationDetails/${returnId.appaId}/${returnId.periodKey}"
 
   def getObligationDataUrl(appaId: String): String =
     s"$adrAccountHost/alcohol-duty-account/obligationDetails/$appaId"
 
+  def getReturnsUrl(returnId: ReturnId): String = {
+    val url = returnsGetReturnUrlFormat.format(Array(regime.toLowerCase, returnId.appaId, returnId.periodKey))
+    s"$returnsHost$url"
+  }
+
   val enrolmentServiceName: String = config.get[String]("enrolment.serviceName")
+
+  val regime: String = config.get[String]("downstream-apis.regimeType")
+
+  private def getConfStringAndThrowIfNotFound(key: String) =
```

Comment:
This method is not needed. Please use getString in ServicesConfig which does the same thing

Reply:
that's not correct

the version that's there requires a default (getConfString) , or doesn't prepend $rootServices (getString)